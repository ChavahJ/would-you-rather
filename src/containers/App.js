// The application works correctly regardless of which user is selected
// Once the user logs in, the home page is shown
// Whenever the user types something in the address bar, the user is asked to log in before the requested page is shown

import { Fragment, useEffect } from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import LoginPage from "./LoginPage";
import Nav from "../components/Nav";
import UserIndicator from "../components/UserIndicator";
import Dashboard from "./Dashboard";
import PollPage from "./PollPage";
import PollCreationPage from "./PollCreationPage";
import LeaderboardPage from "./LeaderboardPage";
import NotFoundPage from "./NotFoundPage";
import { Routes, Route } from "react-router-dom";
import { setAuthedUser } from "../actions/authedUser";

const App = (props) => {
  useEffect(() => {
    console.log(props);
    props.dispatch(handleInitialData());
    const signedIn = localStorage.getItem("authedUser");
    if (signedIn) {
      props.dispatch(setAuthedUser(signedIn));
    }
  }, [props]);

  return (
    <Fragment>
      <main className="main-container">
        {props.authedUser && (
          <header>
            <Nav />
            <UserIndicator />
          </header>
        )}
        <Routes>
          {!props.authedUser && <Route path="/login" element={<LoginPage />} />}
          {props.authedUser && <Route path="/" exact element={<Dashboard />} />}
          {props.authedUser && (
            <Route path="/questions/:question_id" element={<PollPage />} />
          )}
          {props.authedUser && (
            <Route path="/leaderboard" element={<LeaderboardPage />} />
          )}
          {props.authedUser && (
            <Route path="/add" element={<PollCreationPage />} />
          )}
          <Route path="*" element={<NotFoundPage />} />
        </Routes>
      </main>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  authedUser,
});

export default connect(mapStateToProps)(App);
