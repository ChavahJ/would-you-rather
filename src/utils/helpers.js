import { useParams } from "react-router-dom";

export function formatDate(timestamp) {
  const d = new Date(timestamp);
  const time = d.toLocaleTimeString("en-US");
  return time.substr(0, 5) + time.slice(-2) + " | " + d.toLocaleDateString();
}

export const withRouter = (Component) => {
  const ComponentWithRouterProp = (props) => {
    let params = useParams();
    return <Component {...props} router={{ params }} />;
  };
  return ComponentWithRouterProp;
};
