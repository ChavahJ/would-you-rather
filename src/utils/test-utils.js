import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";

import { handleInitialData } from "../actions/shared";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = handleInitialData(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
