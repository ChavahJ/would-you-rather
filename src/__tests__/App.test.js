import { renderWithProviders } from "../utils/test-utils";
import { MemoryRouter } from "react-router-dom";
import App from "../containers/App";

it("renders correctly", () => {
  const component = renderWithProviders(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
