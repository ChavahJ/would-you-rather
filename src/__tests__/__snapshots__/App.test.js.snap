// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main-container"
        data-testid="mainContainer"
      >
        <div
          class="my-5 container"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <h1>
                404 Error
              </h1>
              <p>
                Oops! We can't find that page.
              </p>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <a
                href="/login"
              >
                Log In
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main-container"
      data-testid="mainContainer"
    >
      <div
        class="my-5 container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <h1>
              404 Error
            </h1>
            <p>
              Oops! We can't find that page.
            </p>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <a
              href="/login"
            >
              Log In
            </a>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
